File: sw-expert-academy/D3/오타.java
import java.util.Scanner;
import java.util.stream.Stream;
 
public class Solution {
 
    static long answer;
    static int T;
    static String word;
 
    public static void main(String[] args) {
 
        Scanner sc = new Scanner(System.in);
 
        T = sc.nextInt();
        for (int tc = 1; tc <= T; tc++) {
            // initialize
            answer = 1;
            word = sc.next();
 
            // solution
            int len = word.length();
            for (int i = 0; i < len; i++) {
                // 각 문자에 대해서 좌우 문자를 붙인 부분 문자열 생성
                String substr = word.substring(Math.max(0, i - 1), Math.min(len, i + 2));
                answer *= Stream.of(substr.split("")).distinct().count();
                answer %= 1000000007;
            }
 
            // result
            System.out.printf("#%d %d\n", tc, answer);
        }
    }
}

File: sw-expert-academy/D3/오타.cpp
#include <iostream>
#include <string>
#include <unordered_set>

using namespace std;

int main() {

	int T;
	cin >> T;

	for (int tc = 1; tc <= T; tc++) {
		long long answer = 1;
		string str;
		cin >> str;

		int len = str.length();
		for (int i = 0; i < len; i++) {
			unordered_set<char> hset;
			for (int j = max(0, i - 1); j <= min(len - 1, i + 1); j++) 
				hset.insert(str[j]);
			answer *= hset.size();
			answer %= 1000000007;
		}

		cout << '#' << tc << ' ' << answer << '\n';
	}

	return 0;
}